tests/sources/test_redis.py

```diff
@@ -129,9 +126,8 @@ async def test_validate_config_when_database_type():
 async def test_validate_config_when_database_is_invalid():
     async with create_redis_source() as source:
         source.client.database = ["123"]
-        mocked_client = Mock()
-        mocked_client.validate_database = AsyncMock(return_value=True)
-        with mock.patch("redis.from_url", return_value=mocked_client):
+        source.client.validate_database = AsyncMock(return_value=False)
+        with mock.patch("redis.from_url", return_value=Mock()):
```

Comment:
This patch doesn't work. You should replace all occurrences of `redis.from_url` with `redis.asyncio.from_url` in the test.